modules/imgproc/src/intersection.cpp

```diff
@@ -51,15 +51,15 @@ static int _rotatedRectangleIntersection( const RotatedRect& rect1, const Rotate
 {
     CV_INSTRUMENT_REGION();
 
-    // L2 metric
-    const float samePointEps = std::max(1e-16f, 1e-6f * (float)std::max(rect1.size.area(), rect2.size.area()));
-
     Point2f vec1[4], vec2[4];
     Point2f pts1[4], pts2[4];
```

Comment:
BTW, this code is not performance critical, so probably we could use workaround with `double` computations inside instead of `float`.